src/tls_openssl.c

```diff
@@ -201,6 +201,11 @@ int tls_stop(tls_t *tls)
     int error;
     int ret;
 
+    /* According to OpenSSL.org, we must not call SSL_shutdown(3)
+       if a previous fatal error has occurred on a connection. */
+    if (tls->lasterror == SSL_ERROR_SYSCALL || tls->lasterror == SSL_ERROR_SSL)
+    return 1;
```

Comment:
Please, add 4 spaces before the return statement. This is the block of `if` statement and should be indented accordingly.